raiden-dapp/src/views/SelectHub.vue

```diff
@@ -24,48 +24,51 @@
 
 <script lang="ts">
 import { Component, Mixins } from 'vue-property-decorator';
-import { Token, TokenPlaceholder } from '@/model/types';
+import { Token } from '@/model/types';
 import AddressInput from '@/components/AddressInput.vue';
 import AddressUtils from '@/utils/address-utils';
 import NavigationMixin from '@/mixins/navigation-mixin';
 import Divider from '@/components/Divider.vue';
 import TokenInformation from '@/components/TokenInformation.vue';
 import ActionButton from '@/components/ActionButton.vue';
-import { mapState } from 'vuex';
+import { mapGetters, mapState } from 'vuex';
 
 @Component({
   components: { TokenInformation, Divider, AddressInput, ActionButton },
   computed: {
-    ...mapState(['defaultAccount'])
+    ...mapState(['defaultAccount']),
+    ...mapGetters({
+      getToken: 'token'
+    })
   }
 })
 export default class SelectHub extends Mixins(NavigationMixin) {
-  token: Token = TokenPlaceholder;
   defaultAccount!: string;
+  getToken!: (address: string) => Token;
 
   partner: string = '';
   valid: boolean = false;
 
+  get token(): Token {
+    const { token: address } = this.$route.params;
+    return this.getToken(address) || ({ address } as Token);
+  }
+
   selectHub() {
     this.navigateToOpenChannel(this.token.address, this.partner);
   }
 
   async created() {
-    const { token } = this.$route.params;
-    if (!AddressUtils.checkAddressChecksum(token)) {
+    const { token: address } = this.$route.params;
+    if (!AddressUtils.checkAddressChecksum(address)) {
       this.navigateToHome();
       return;
     }
 
-    let tokenInfo = this.$store.getters.token(token);
-    if (!tokenInfo) {
-      tokenInfo = await this.$raiden.getToken(token);
-    }
+    await this.$raiden.fetchTokenData([address]);
```

Comment:
Hmm, straight up calling `fetchTokenData` may be a problem. It'll call `raiden.getTokenInfo` & `raiden.getTokenBalance`, which will then fail if the token isn't in state yet (which happens e.g. on first launch, if the user is redirected directly to a SelectHub route from the Wizard). `$store.state.tokens` is populated by the async fetching of the tokens. I think ideally you should `watch` for a few seconds if the token shows up in your state (from the subscription in the service), and then if so, `fetchTokenData`, or else navigate to home if it times out. Or possibly include this check/watch/timeout directly in service's `fetchTokenData` and handle the rejection here.

Reply:
This is not a problem at all during the first lunch due to the way everything is setup.

A route component (`SelectHub`, `Payment` etc) won't be created until the `loadComplete` mutation is executed since the `<router-view>` is not shown until the initialization is done.

Before committing `loadComplete` we await the `getTokenList` and update the tokens with the placeholders already, so by the time the router view is visible and thus the component is created the placeholders are already in the vuex state.

I have also verified that this works by hand.

Reply:
This works well if the token on the route being accessed is already in SDK's state, so it gets pulled in by `getTokenList`, but on first run (no state), this method resolves as soon as the first token show up, the rest are lazy-loaded, but if the route is accessing one of the last, it won't be yet on the `getTokenList`, nor the `$store.state.tokens`, and `fetchTokenData` may fail.
But nvm, if it works by hand, I may be missing something that's going synchronous enough there. Thank you